### PR TITLE
[JENKINS-33463] Support BuildResultTriggerPlugin

### PR DESCRIPTION
**Pull request by ldez**
- https://wiki.jenkins-ci.org/display/JENKINS/BuildResultTrigger+Plugin

Fixes [JENKINS-33463](https://issues.jenkins-ci.org/browse/JENKINS-33463)

### CODE REVIEW

File: job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/triggers/TriggerContext/buildResult.groovy [hunk 1]
```diff
@@ -0,0 +1,25 @@
+job('example') {
+    triggers {
+        buildResult('H/10 * * * *') {
+            combinedJobs()
+            triggerInfo('My job 1') {
```
Comment (daspilker) [inline job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/triggers/TriggerContext/buildResult.groovy]: Do we really need the nested closure? `triggerInfo(String job, BuildResult... buildResult)` would reduce the complexity.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/triggers/BuildResultTriggerContext.groovy [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    boolean combinedJobs
+    final List<BuildResultTriggerInfoContext> buildResultInfos = []
+
+    BuildResultTriggerContext(String cron = null) {
+        if (cron) {
+            this.crontab = cron
+        }
+    }
+
+    void cron(String cron) {
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/triggers/BuildResultTriggerContext.groovy]: Add GroovyDoc for all DSL methods

File: job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/triggers/TriggerContext/buildResult.groovy (hunk 1 above)
Comment (ldez) [inline job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/triggers/TriggerContext/buildResult.groovy]: My first approch is like you say, but I change my implementation for respect mandatory rules from the plugin.

Comment (ldez) [inline job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/triggers/TriggerContext/buildResult.groovy]: If you think `triggerInfo(String job, BuildResult... buildResult) ` is a better design, I refactor.

Comment (daspilker) [inline job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/triggers/TriggerContext/buildResult.groovy]: The first (and undocumented) rule is "keep it simple" :smirk:

The optional vs. mandatory options design rule should avoid long parameter list with optional parameters, e.g. `foo(String one, int two = 0, String three= "bar", boolean four = false)`. I think this case is different, so we can make an exception. Please do the refactoring.

Comment (ldez) [inline job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/triggers/TriggerContext/buildResult.groovy]: I when to say mandatory rules, I talk about BuilldResult Plugin:
- `triggerInfo` can be defined without defined `BuildResult` and no `BuildResult` is defined in this case.
- I defined a `BuildResult` default state only when `buildResult()` is calling.
`BuildResult... buildResults = [BuildResult.SUCCESS]`

But I do the refactoring.

**Comment by ldez** [comment]
refactor done!

**Pull request merged**